Site/SiteAnalyticsModule.php

```diff
@@ -146,11 +175,98 @@ public function hasAnalytics()
 		return (
 			$this->hasGoogleAnalytics() ||
 			$this->hasFacebookPixel() ||
+			$this->hasTwitterPixel() ||
 			$this->hasBingUET()
 		);
 	}
 
 	// }}}
+	// {{{ public function displayAnalytics()
+
+	public function displayAnalytics()
+	{
+		$this->displayGoogleAnalytics();
+		$this->displayFacebookPixelImage();
+		$this->displayTwitterPixelImages();
+		$this->displayBingUETImage();
+	}
+
+	// }}}
+	// {{{ public function getInlineJavascript()
+
+	public function getInlineJavascript()
+	{
+		$js = '';
+
+		if ($this->hasFacebookPixel()) {
+			$js.= $this->getFacebookPixelInlineJavascript();
+		}
+
+		if ($this->hasBingUET()) {
+			$js.= $this->getBingUETInlineJavascript();
+		}
+
+		if ($this->hasGoogleAnalytics()) {
+			$js.= $this->getGoogleAnalyticsInlineJavascript();
+		}
+
+		if ($this->hasTwitterPixel()) {
+			$js.= $this->getTwitterPixelInlineJavascript();
+		}
+
+		return $js;
+	}
+
+	// }}}
+	// {{{ protected function initOptOut()
+
+	protected function initOptOut()
+	{
+		$cookie_module = null;
+
+		if ($this->app->hasModule('SiteCookieModule')) {
+			$cookie_module = $this->app->getModule('SiteCookieModule');
+
+			if (isset($cookie_module->AnalyticsOptOut)) {
+				$this->analytics_opt_out = true;
+			}
+		}
+
+		if (isset($_GET['AnalyticsOptIn'])) {
+			$this->analytics_opt_out = false;
+			if ($cookie_module === null) {
+				$e = new SiteException('Attempting to remove Analytics Opt '.
+					'Out Cookie with no SiteCookieModule available.');
+
+				$e->processAndContinue();
+			} else {
+				$cookie_module->removeCookie('AnalyticsOptOut');
+			}
+		}
+
+		// Opt Out trumps opt in if you include them both flags in your query
+		// string for some reason.
+		if (isset($_GET['AnalyticsOptOut'])) {
```

Comment:
`SiteApplication::initVar()` would be better here than accessing $_GET directly.